src/biokbase/narrative/tests/test_auth.py

```diff
@@ -0,0 +1,198 @@
+import json
+import os
+
+import pytest
+from requests import HTTPError
+
+from biokbase.auth import (
+    TokenInfo,
+    get_agent_token,
+    get_auth_token,
+    get_display_names,
+    get_token_info,
+    init_session_env,
+    set_environ_token,
+    validate_token,
+)
+from biokbase.narrative.common.url_config import URLS
+from biokbase.narrative.common.util import kbase_env
+
+AUTH_URL = URLS.auth + "/api/V2/"
+
+unauth_error = {
+    "error": {
+        "httpcode": 401,
+        "httpstatus": "Unauthorized",
+        "appcode": 10020,
+        "apperror": "Invalid token",
+        "message": "10020 Invalid token",
+        "callid": "7024216894157398",
+        "time": 1691184318526,
+    }
+}
+
+bad_request_error = {
+    "error": {
+        "httpcode": 400,
+        "httpstatus": "Bad Request",
+        "appcode": 10010,
+        "apperror": "No authentication token",
+        "message": "10010 No authentication token: No user token provided",
+        "callid": "7036685833684377",
+        "time": 1691184644451,
+    }
+}
+
+
+@pytest.fixture
```

Comment:
would love to move all the narrative tests over to pytest -- in particular, parameterisation and mocking are so much nicer than in unittest.

Reply:
Yes! I started an effort toward that a long time ago - the test runner is pytest, for example - but didn't get very far. I figure, as long as new tests are being made, start converting them over.

I didn't go as far as making a conftest.py and putting fixtures in there, as I want to keep this a fairly lightweight PR. I think that's worth doing as future work and introducing a reasonable mocking framework.

Reply:
💯  Begone, foul `@unittest.mock`!